delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resolvers/ResolverResolution.scala

```diff
@@ -143,9 +144,19 @@ final class ResolverResolution[R](
       projectRef: ProjectRef,
       resolver: InProjectResolver
   ): IO[ResolverResolutionResult[R]] =
-    fetch(ref, projectRef).map {
-      case None => ResolverReport.failed(resolver.id, projectRef -> ResolutionFetchRejection(ref, projectRef)) -> None
-      case s    => ResolverReport.success(resolver.id, projectRef)                                             -> s
+    for {
+      resourceOpt <- fetch(ref, projectRef)
+      result      <- resourceOpt.traverse(runDeprecationCheck(ref, projectRef, _))
+    } yield {
+      result match {
+        // The resource has not been found in the project
+        case None        => ResolverReport.failed(resolver.id, projectRef -> ResolutionFetchRejection(ref, projectRef)) -> None
+        // The resource exists but the deprecation check is positive so we reject it
+        case Some(true)  =>
+          ResolverReport.failed(resolver.id, projectRef -> ResourceIsDeprecated(ref.original, projectRef)) -> None
+        // The resource has been successfully resolved
+        case Some(false) => ResolverReport.success(resolver.id, projectRef)                                             -> resourceOpt
+      }
```

Comment:
I think by using meaningful variable and method names you don't really need the comments here

Reply:
Something like this? 

scala
```
    fetch(ref, projectRef).flatMap {
      case None => IO.pure(ResolverReport.failed(resolver.id, projectRef -> ResolutionFetchRejection(ref, projectRef)) -> None)
      case Some(resource) => shouldExcludeDueToDeprecation(ref, projectRef, resource)).map {
        case Exclude => ResolverReport.failed(resolver.id, projectRef -> ResourceIsDeprecated(ref.original, projectRef)) -> None
        case Include => ResolverReport.success(resolver.id, projectRef) -> Some(resource)
      }
    }
```

Reply:
It does not look really clearer and it will have an impact on the cross-project resolver side

Reply:
I don't fully understand the impact the change would have on cross-project resolver 🤔

Reply:
Obviously this is all a matter of opinion, but the ways in which I would see it as objectively clearer:
- No comments needed
- Boolean matching removed for clearer case objects
- `resourceOpt` is only used when it is Some, this is fixed
- shouldExcludeDueToDeprecation is clearer than runDeprecationCheck as it is explaining the check
- Traversal is maybe not a big deal but it is a complication

I think we can move on for this PR, but the things I have mentioned are things which in general make the codebase we have difficult to understand/navigate. They are all small things by themselves, but they add up. 

I'm just posting this to attempt to express how I feel about it, I think it's fine to disagree